rfc/0011-audit-checks.md

```diff
@@ -113,6 +113,370 @@ resources of a given Namespace.
 
 That's because:
 
+* Once all the policies and involved resources are audited, the
+  `PolicyReport` audit is going to be overwritten with the results of the audit
+
+Users can still get live data about the rejections/approvals by looking both
+at Prometheus data and Open Telemetry traces.
+
+We should focus on making the process of creating a `PolicyReport` as fast as
+possible. The reports are recreated on a regular basis. If a user wants a fresh
+`PolicyReport` without having to wait for the next scheduled run, we can
+provide a way for the user to trigger this operation.
+
+Finally, if we really have to implement live update of `PolicyReport`, we can
+come up with an alternative architecture that will allow for that. For example,
+we could have all the write operations of `PolicyReport` to go through a
+dedicated "writer" service.
+
+## Creating a `PolicyReport` and `ClusterPolicyReport` objects
+
+The following section describes how  we will produce a `PolicyReport` for
+namespaced resources and `ClusterPolicyReport` for cluster-wide resources
+describing the compliance status.
+
+### Creating a `ClusterPolicyReport` object
+
+A `ClusterPolicyReport` compiles the evaluation results for cluster-wide
+resources. The creation process entails several steps:
+
+#### Step 1: find relevant policies
+
+The `ClusterAdmissionPolicy` monitors activities across all namespaces and at the
+cluster level. The initial step involves collecting these policies and
+categorizing them based on the types of cluster-wide resources they examine.
+The `GroupVersionResource` identifier helps in recognizing the specific
+Kubernetes resource types involved.
+
+**Note**: Only policies that review CREATE events are taken into account.
+
+#### Step 2: fetch Kubernetes resources to be audited
+
+After determining the policies for each resource type, the next task is to
+gather the instances of these resources present in the cluster. This is done by
+querying the API server, often using pagination to manage the data efficiently.
+
+Post-retrieval, the audit scanner assesses whether each resource aligns with
+the criteria set out in the policies, specifically focusing on the labels
+required by the policy. If a resource fits the criteria, the scanner simulates
+a `CREATE` event, incorporating the audited Kubernetes object as `request.object`
+within this event.
+
+### Step 3: performing evaluations
+
+Initially, the evaluation is executed through an HTTP request directed at the
+Policy Server that enforces the policy. This step is crucial for determining
+the compliance of each resource with the established cluster policies.
```

Comment:
I would add that we are cylcing all the resources and auditing the resources one by one against the policies they are targeted by.